Code/GraphMol/Canon.cpp

```diff
@@ -661,6 +670,9 @@ void dfsBuildStack(ROMol &mol, int atomIdx, int inBondIdx,
         continue;
       }
       unsigned long rank = ranks[otherIdx];
```

Comment:
This is doing work it doesn't need to do. I think it makes sense to replace this piece of code:
```
      unsigned long rank = ranks[otherIdx];
      if (theBond->getOwningMol().getRingInfo()->numBondRings(
              theBond->getIdx())) {
        if (!bondSymbols) {
          rank += static_cast<int>(MAX_BONDTYPE - theBond->getBondType()) *
                  MAX_NATOMS * MAX_NATOMS;
        } else {
          const std::string &symb = (*bondSymbols)[theBond->getIdx()];
          boost::uint32_t hsh = gboost::hash_range(symb.begin(), symb.end());
          rank += (hsh % MAX_NATOMS) * MAX_NATOMS * MAX_NATOMS;
        }
      }
```
with this:
```
      unsigned long rank = ranks[otherIdx];
      if (!doRandom) {
        if (theBond->getOwningMol().getRingInfo()->numBondRings(
                theBond->getIdx())) {
          if (!bondSymbols) {
            rank += static_cast<int>(MAX_BONDTYPE - theBond->getBondType()) *
                    MAX_NATOMS * MAX_NATOMS;
          } else {
            const std::string &symb = (*bondSymbols)[theBond->getIdx()];
            boost::uint32_t hsh = gboost::hash_range(symb.begin(), symb.end());
            rank += (hsh % MAX_NATOMS) * MAX_NATOMS * MAX_NATOMS;
          }
        }
      } else {
        // randomize the rank
        seed = std::rand();
      }
```
instead of what you've done.